packages/ember-metal/lib/computed.js

```diff
@@ -460,13 +457,13 @@ ComputedPropertyPrototype._set = function computedPropertySet(obj, keyName, valu
 /* called before property is overridden */
 ComputedPropertyPrototype.teardown = function(obj, keyName) {
   var meta = metaFor(obj);
-
-  if (meta.cache) {
-    if (keyName in meta.cache) {
+  let cache = meta.readableCache();
+  if (cache) {
+    if (keyName in cache) {
```

Comment:
if I recall, the cache will never have an undefined member representing the value of undefined, rather it will use the UNDEFINED sentinel value. This means, we can likely just do (the unfortunately faster) `if (cache[keyName] !== undefined)`